src/backend/blockchain/src/blockchain.py

```diff
@@ -61,5 +61,38 @@ def initialize_chain(self):
     def get_last_block(self):
         return self.chain[-1]
 
+    def build_chain_from_peer_response(self, data):
+        try:
+            new_chain = []
+            chain = data["chain"]
+            for block in chain:
+                block = json.loads(block)
+                new_chain.append(Block.from_json(block))
+            
+            if new_chain:
+                self.chain = new_chain
+        
+        except KeyError as e:
+            print("LOG: Incorrect Request, Keyerror:", str(e))
```

Comment:
I don't think we need these exception checks as they don't really do anything other than log. If we had some sort of error handling/recovery I would see the value but otherwise, I think this one and the three others in this file can be removed

Reply:
I have the error checks to not make the blockchain crash, because this is done on startup I am worried an error being thrown will make the entire container crash. which is why not we would still need the `try` & `except` block, but we can just have a `pass` in except to not do anything. What do you think?

Reply:
I think it'd be better to remove the try here return an exception or something and then in the route, where this is being called, use the try instead and return a corresponding exception if the error occurs. This would match how we do error handling in other endpoints

Reply:
Yup that would make sense, this is not used in the route but in the query_peers method so will just keep the try except there

Reply:
also, we don't need to handle exceptions for replication, it fails no problem which is why no error handling in routes :)